cassovary-core/src/main/scala/com/twitter/cassovary/graph/MemoryMappedDirectedGraph.scala

```diff
@@ -122,190 +159,164 @@ object MemoryMappedDirectedGraph {
+        }
+        val offsetsBuffer = binaryGraphChannel.map(MapMode.READ_WRITE, chunkOffsetsStart,
+          8 * (nodeCountInChunk + 1)) // The + 1 is for the edge case described below
+
+        for (offset <- neighborOffsets) {
+          offsetsBuffer.putLong(offset)
+        }
+        // Edge case: For the last chunk, we need to write the final offset, in order to store
+        // the nth node's out-degree (and in-degree).
+        if (nodesPerChunk * (chunkIndex + 1) >= nodeCount) {
+          offsetsBuffer.putLong(finalOffset)
+        }
+        offsetsBuffer.force()
+      }
     }
-    reader.toArrayBasedDirectedGraph()
-  }
 
-  def main(args: Array[String]): Unit = {
-    var startTime = System.currentTimeMillis()
-    val testNodeId = 1
-    val graphName = args(1)
-    if (args(0) == "readAdj") {
-      val graph = readAdjacencyListToGraph(graphName)
-      println(s"outneighbors of node $testNodeId: " + graph.getNodeById(testNodeId).get.outboundNodes())
-      val loadTime = (System.currentTimeMillis() - startTime) / 1000.0
-      println(s"Time to read adj graph: $loadTime")
-
-      val binaryFileName = graphName.substring(0, graphName.lastIndexOf(".")) + ".dat"
-      startTime = System.currentTimeMillis()
-      MemoryMappedDirectedGraph.graphToFile(graph, new File( binaryFileName))
-      val writeTime = (System.currentTimeMillis() - startTime) / 1000.0
-      println(s"Time to write binary graph: $writeTime")
-    } else if (args(0) == "readBin") {
-      val graph = new MemoryMappedDirectedGraph(new File(graphName))
-      println(s"outneighbors of node $testNodeId: " + graph.getNodeById(testNodeId).get.outboundNodes())
-      val loadTime = (System.currentTimeMillis() - startTime) / 1000.0
-      println(s"Time to read binary graph: $loadTime")
-    } else if (args(0) == "readEdges") {
-      val binaryFileName = graphName.substring(0, graphName.lastIndexOf(".")) + ".dat"
-      MemoryMappedDirectedGraph.edgeFileToGraph(new File(graphName), new File(binaryFileName))
-      val writeTime = (System.currentTimeMillis() - startTime) / 1000.0
-      println(s"Time to convert graph: $writeTime")
-    } else if (args(0) == "readEdgesSorted") {
-      assert(args.size == 4, "arguments: readEdgesSorted <edge file sorted by id1> " +
-        "<edge file sorted by id2> <binary output file>")
-      val binaryFileName = args(3)
-      MemoryMappedDirectedGraph.sortedEdgeFilesToGraph(
-        new File(args(1)),
-        new File(args(2)),
-        new File(binaryFileName))
-      val writeTime = (System.currentTimeMillis() - startTime) / 1000.0
-      println(s"Seconds to convert graph: $writeTime")
-    } else {
-      println("unexpected command")
-    }
+    log(s"wrote $halfEdgeCount half-edges")
+    binaryGraphOutput.close()
+    binaryGraphChannel.close()
+    tempFilesById1 foreach (_.delete())
+    tempFilesById2 foreach (_.delete())
+    log("finished writing graph at " + new Date())
   }
```

Comment:
I just missed the fact that they were being deleted. We don't need to do it optionally -- I thought you were keeping them around for some use case.